cartridges/int_algolia/cartridge/scripts/algolia/steps/algoliaProductDeltaIndex.js

```diff
@@ -385,50 +377,49 @@ exports.process = function(cpObj, parameters, stepExecution) {
             if (product.master) {
                 let variationModel = product.getVariationModel();
                 let variationAttribute = variationModel.getProductVariationAttribute(VARIATION_ATTRIBUTE_ID);
-                if (!variationAttribute) {
-                    logger.info('No ' + VARIATION_ATTRIBUTE_ID + ' attribute found for product ' + product.ID + ', will generate variant records');
-                    // TODO: move in a shared function, the code is the same than the "attributesComputedFromBaseProduct" condition below
-                    let recordsPerLocale = jobHelper.generateVariantRecords({
-                        masterProduct: product,
-                        locales: siteLocales,
-                        attributeList: attributesToSend,
-                        nonLocalizedAttributes: nonLocalizedAttributes,
-                        attributesComputedFromBaseProduct: attributesComputedFromBaseProduct,
-                        fullRecordUpdate: fullRecordUpdate,
-                    });
-                    for (let l = 0; l < siteLocales.size(); ++l) {
-                        let locale = siteLocales[l];
-                        let indexName = algoliaData.calculateIndexName('products', locale);
-                        let records = recordsPerLocale[locale];
-                        processedVariantsToSend = records.length;
-
-                        records.forEach(function (record) {
-                            if (INDEX_OUT_OF_STOCK || record.in_stock) {
-                                algoliaOperations.push(new jobHelper.AlgoliaOperation(baseIndexingOperation, record, indexName));
-                            } else {
-                                algoliaOperations.push(new jobHelper.AlgoliaOperation(deleteIndexingOperation, { objectID: record.objectID }, indexName));
-                            }
-                        });
 
-                        // Delete records of variants that don't match the filter anymore.
-                        let variants = product.variants;
-                        if (variants && variants.size() > 0) {
-                            for (let v = 0; v < variants.size(); ++v) {
-                                let variant = variants[v];
-                                if (!productFilter.isInclude(variant)) {
-                                    algoliaOperations.push(new jobHelper.AlgoliaOperation(deleteIndexingOperation, { objectID: variant.getID() }, indexName));
-                                }
-                            }
+                // masters that DON'T have the specified variation attribute -- treat them as regular masters
+                // @TODO: same logic as for masters in the MASTER_LEVEL case, abstract away and refactor
+                if (!variationAttribute) {
+                    logger.info('Specified variation attribute "' + VARIATION_ATTRIBUTE_ID + '" not found for master product "' + product.ID + '", falling back to master-type record.');
```

Comment:
```suggestion
                    logger.info('Specified variation attribute "' + VARIATION_ATTRIBUTE_ID + '" not found for master product "' + product.ID + '", falling back to master-level record.');
```

Reply:
There are 3 types of final records.
In this condition, we are building a master-level record, i.e. a record with all variants present in the `variants` array, variation-group-level records are built outside of this condition (your comment above states this explicitly).
So there is no reason for the log to not be explicit.